streams/src/test/java/org/apache/kafka/streams/processor/internals/StateDirectoryTest.java

```diff
@@ -266,4 +268,51 @@ public void shouldUnlockGlobalStateDirectory() throws Exception {
         }
     }
 
+    @Test
+    public void shouldNotLockStateDirLockedByAnotherThread() throws IOException, InterruptedException {
+        final TaskId taskId = new TaskId(0, 0);
+        final Thread thread = new Thread(new Runnable() {
+            @Override
+            public void run() {
+                try {
+                    directory.lock(taskId, 1);
+                } catch (IOException e) {
+                    //
```

Comment:
Or does `IOException` and "didn't get lock" correlate and the test would fail in L286? For this case, we should at least call `printStackTrace()` and `flush` to make debugging possible.